### DISCUSSION TITLE
French layout

### DISCUSSION THREAD

**Discussion by treeent** [8VIM/8VIM#561] [Language Layout]
Hello,

New here. This keyboard seems to be quite a game changer, thanks @flide!

As my primary language is French, I´m really interested in the ways to customize 8vim. (and will be making many spelling/grammar/idiomatic mistakes in my posts 😳[^1]) Obviously, default layout is not optimized for French --I totally understand and accept that: it´s not a French-centered project. 

I saw I can select alternative layouts in the config menu. But, as far as I understand, the currently available (6-layers) French layout only differs from the (1-layer) English one by the addition of the diacritics/special characters layers. These extra layers really are helpful, but its main/top layer is still the same and, thus, is not optimized for French. [^2]

I already tested the side-loading of @martin-braun .yaml file. It works but
- even if it scores better for French than the original layout (as I could read in another thread), it is not intended to be optimized for this language
- it seems to install only a top layer (I cannot access the other layers), even if I read that this layout is one of the most extensive one

In #254, @Glitchy-Tozier suggests multiple top-layer layouts for French (and other languages). But I cannot find how to install them as a layout or, even better, as a main layer on top of the diacritics layers of the official French layout.

Could any of you help me? After testing what´s already available, I´ll try to better the French layout ... with your help. 😏 

[^1]: I hope I´ll still be understandable and that I won´t write anything offensive ... just tell me if need be😉
[^2]: BTW, can anyone explains me the difference between _extra layer_ and _hidden layer_ [^3]
[^3]: BTW, is there specific way to talk about 1-layer-only layout and many-layers layout? Something like _surface layout_ and _deep layout_?

**Comment by martin-braun** [comment]
The file that you found from me is a German/English optimized layout with all possible latin dialect characters, number shortcuts and - added by me - additional special characters on shorter keymaps that don't follow the FOLI center rule ("first out, last in" - yes I just invented that term), which are faster to write - but staggering in the flow, because you have to lift the finger to write them.

@kjoetom is actually the person who came up with the optimized layout that I originated my layout from, he did the most work on it, I just added some additional stuff to it that is important for my personal work-flow. I'm sure he likes to point you in the right direction in regards of how you can do the same that he did, but for French/English or just French.

---

In regards of [2]: An extra layer is a later invention of 8VIM that can be swapped to. I don't really know about it, because older versions didn't have it and we built our layouts without such feature. We exclusively use hidden layers to get additional characters. Hidden layers are just characters that you cannot see on the front, because they require additional movement beyond the presentation (i.e. moving one cell back before entering `C` would enter a hidden layer that prints another character that is defined in the keymaps, but not visible in the UI, hence hidden). Our layout is full of hidden layers as @kjoetom tried to cover all latin letters possible.

**Comment by Glitchy-Tozier** [comment]
Hi, just a quick note: While the layouts proposed in #254 are way better than the original one, there have been improvements to the optimizer. The most important one is that now we can optimize a layout for multiple languages (for example 65% French, 35% English).

Should you decide to optimize such a layout and require any help, hit me up :)

**Reply by treeent** [reply]
Thanks @Glitchy-Tozier
